### PR TITLE
add topicname args to navigation launch

### PR DESCRIPTION
**Pull request by sktometometo**
I have added topic name arguments to navigation launch files so that navigation topics can be remapped to other topics. ( like remapping /odom to /odom_visual )

### CODE REVIEW

**Review by cjds — APPROVED** [review]
SRTM

**Review by cjds — APPROVED** [review]
SRTM

File: fetch_navigation/launch/fetch_nav.launch [hunk 1]
```diff
@@ -11,6 +11,12 @@
   <arg name="move_base_include" default="$(find fetch_navigation)/launch/include/move_base.launch.xml" />
   <arg name="amcl_include" default="$(find fetch_navigation)/launch/include/amcl.launch.xml" />
 
+  <!-- set topics -->
+  <arg name="scan_topic"     default="base_scan" />
```
Comment (narora1) [inline fetch_navigation/launch/fetch_nav.launch]: remove tab

File: fetch_navigation/launch/freight_nav.launch [hunk 2]
```diff
@@ -11,6 +11,12 @@
   <arg name="move_base_include" default="$(find fetch_navigation)/launch/include/move_base.launch.xml" />
   <arg name="amcl_include" default="$(find fetch_navigation)/launch/include/amcl.launch.xml" />
 
+  <!-- set topics -->
+  <arg name="scan_topic"     default="base_scan" />
```
Comment (narora1) [inline fetch_navigation/launch/freight_nav.launch]: remove tab

**Review by narora1 — CHANGES_REQUESTED** [review]

**Comment by erelson** [comment]
A question for others in general (incl. @cjds @narora1 ): Does it make sense to be parameterizing everything in the base launch files?

Where I'm coming from: My impression is that the launch files originally captured best practices (5 years ago), but I'm not personally familiar with other robots' equivalent launch files or changes in practice in the past 5 years. I am wondering if these changes (including similar in previous PRs) match the spirit of current best/common practices.

That said, this generally LGTM 👍 . I just could imagine a fully parameterized launch file being a bit less readable...

**Comment by mikeferguson** [comment]
@erelson I think it makes sense that as you have more people doing more use-cases, you'll end up with a few more parameters. I can certainly understand scan and odom getting parameterized (since you might install a better/different set of sensors).

Parameterizing the default map seems kind of unnecessary (I'm not aware of any other robot platform that uses a default map not named "map" - it's pretty much the standard). Same goes for cmd_vel - while different robots may have a different name for this, the entire fetch/freight research system uses a consistent naming convention throughout the real robot config (that's apps like nav here, as well as drivers/bringup) and the simulation.

**Comment by knorth55** [comment]
In our lab, `map` can be changed when a robot moves to different level and manages each level map as `/map/1f`, `/map/2f` and so on.
Also, `cmd_vel_topic` can be changed when we make base controller to do motion planning with base.

**Comment by knorth55** [comment]
kindly ping.
can we need to modify this PR? or can you merge this PR?

**Comment by erelson** [comment]
Thanks for the ping. Merged both indigo and melodic PRs :+1:

**Pull request merged**